lib/cocoapods/sources_manager.rb

```diff
@@ -68,8 +68,10 @@ def create_source_with_url(url)
       #         The URL of the source.
       #
       def cdn_url?(url)
-        url =~ %r{^https?:\/\/} &&
-          REST.head(url + '/all_pods.txt').ok?
+        if url =~ %r{^https?:\/\/}
+          response = REST.get(url + '/CocoaPods-version.yml')
```

Comment:
will this follow redirects or we no longer care about it anymore?

Reply:
no redirects here. Just changed head to get request and checking response body

Reply:
What I am asking though is whethere we _want_ to follow re-directs or not?

Reply:
To be honest, I don't know. I assuming if initial logic has not possibility to redirect then it should not be changed. I am interesting in adding basic auth possibility. But seems it is better to postpone such fix a little bit. As far Typhoeus lib will be added in nearest future and it supports auto login with .netrc therefore most likely I will create PR to use here Typhoeus if @igor-makarov will not do it before me. If I understood him properly he can refactor `http` class from core repo to use Typhoeus and in such case change here will be very simple. This PR appeared because I have promised add logic to check response body in addition to 200 error code checking.